src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -133,7 +104,7 @@ public VisitArgumentResult(VisitResult visitResult, Optional<LocalState> stateFo
         /// <summary>
         /// Return statements and the result types from analyzing the returned expressions. Used when inferring lambda return type in MethodTypeInferrer.
         /// </summary>
-        private readonly ArrayBuilder<(BoundReturnStatement, TypeWithAnnotations)> _returnTypesOpt;
+        private ImmutableArray<(BoundReturnStatement, TypeWithAnnotations)> _returnTypesOpt;
```

Comment:
>private ImmutableArray<(BoundReturnStatement, TypeWithAnnotations)> _returnTypesOpt; [](start = 8, length = 84)

I do not believe there is any advantage to change the type of this field to immutable array.